main.c

```diff
@@ -135,6 +135,10 @@ main(int argc, char *argv[])
 			verbose = 1;
 			break;
 		GETOPT_OPT("-P"):
+			if (passphrase_entry != PASSPHRASE_UNSET) {
```

Comment:
This is technically a change of behaviour, rather than a refactoring.  But I think it's small enough to not need a separate PR.